source/core/mesh_controller.cc

```diff
@@ -0,0 +1,51 @@
+//
+// Created by luckab on 12/07/23.
+//
+
+#include <core/mesh_controller.h>
+
+
+using namespace std;
+double
+MeshController::calculate_coarsening_factor(int current_number_of_cell)
```

Comment:
To change with refactor
```suggestion
MeshController::calculate_coarsening_factor(int current_number_of_cells)
```